server/rooms.ts

```diff
@@ -782,10 +796,10 @@ export abstract class BasicRoom {
 		this.users[user.id] = user;
 		if (joining) {
 			this.reportJoin('j', user.getIdentityWithStatus(this.roomid), user);
-			if (this.settings.staffMessage && user.can('mute', null, this)) {
+			if (user.can('mute', null, this)) {
 				this.sendUser(
 					user,
-					`|raw|<div class="infobox">(Staff intro:)<br /><div>${this.settings.staffMessage.replace(/\n/g, '')}</div></div>`
+					this.getStaffIntroMessage(user)
```

Comment:
This is good, but will send an empty message if a room has neither a staff message nor pending approvals.

I think the `user.can('mute', null, this)` check should be moved inside `getStaffIntroMessage`, and this line should have a check that if it's not there, it's not sent.